core/src/main/java/org/apache/calcite/sql/SqlDialect.java

```diff
@@ -532,12 +535,29 @@ public SqlNode rewriteSingleValueExpr(SqlNode aggCall) {
    *
    * @param node The SqlNode representing the expression
    * @param nullsFirst <code>true</code> if nulls should come first, <code>false</code> otherwise
+   * @param desc <code>true</code> if the sort direction is RelFieldCollation.Direction.DESCENDING
+   *             or RelFieldCollation.Direction.STRICTLY_DESCENDING
    * @return A SqlNode for null direction emulation or <code>null</code> if not required
    */
-  public SqlNode emulateNullDirection(SqlNode node, boolean nullsFirst) {
+  public SqlNode emulateNullDirection(SqlNode node, boolean nullsFirst, boolean desc) {
     return null;
   }
 
+  protected SqlNode emulateNullDirectionForLowNulls(
```

Comment:
How about considering the `nullCollation` already present in `SqlDialect`?
Rename this method to `emulateNullDirectionWithIsNull` and skip the  null emulation based on the `nullCollation` and the given `nullsFirst` and `desc` settings. It's not relevant for your purposes but is a nice and small generalization that we could use later for other DBMS.